cityofaustin/atd-vz-data
# 9899 add location notes

## Associated issues

Closes cityofaustin/atd-data-tech/issues/9899
Closes cityofaustin/atd-data-tech/issues/9900
Closes cityofaustin/atd-data-tech/issues/9901

This PR creates a Notes section in the Location details that has the same functionality as the Crash details Notes feature. Users with edit & admin permissions should be able to read, add, edit, and delete notes. Users with only read permissions should only be able to read existing notes.
 
## Testing
**URL to test:** 

This can be tested locally or as a Netlify preview (https://deploy-preview-1090--atd-vze-staging.netlify.app/#/dashboard)

**Steps to test:**

1. Go to Locations tab.
2. Click on Location ID 644AF55CFC (https://deploy-preview-1090--atd-vze-staging.netlify.app/#/locations/644AF55CFC)
3. New Notes section should be underneath aerial map and details components.
4. If you have edit permissions, add a new note using add button. New note should appear under the add note row.
5. The edit button (pencil) should only appear for notes you created (updated by column).
6. Click edit button on a note you created then cancel (X button). Note should be unchanged.
7. Click edit button, change nothing, then save (check button). Note should be unchanged.
8. Click edit button, change the note, then save. Note should be updated.
9. Delete your note.
10. Go to Location ID 8FCA19BF67 or 0A4FCA4EA6 to see other Notes sections with dummy data, repeat steps if you wish!

---
#### Ship list

Note: For deployment, the following steps will need to be done in the Production Hasura console:
- [ ] Rename "notes" table "crash_notes"
- [ ] Add a location notes table called "location_notes"
- [ ] Add date, user_email, created_at, updated_at, text, id, location id columns to "location_notes" with the same parameters as staging console.
